fahrschein-e2e-test/src/test/java/org/zalando/fahrschein/e2e/NakadiClientEnd2EndTest.java

```diff
@@ -78,17 +78,8 @@ public class NakadiClientEnd2EndTest extends NakadiTestWithDockerCompose {
         objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
     }
 
-    public final RequestFactory requestFactory;
-
-    private NakadiClient nakadiClient;
-
-    public NakadiClientEnd2EndTest(RequestFactory requestFactory, String testName) {
-        this.requestFactory = requestFactory;
-    }
-
-    @Before
-    public void setUpNakadiClient() {
-        nakadiClient = NakadiClient
+    private NakadiClient setUpNakadiClient(RequestFactory requestFactory) {
```

Comment:
Wouldn't it make sense to still have a fresh client per test?

Reply:
we still do it. We're calling it from within the test methods now.